### PR TITLE
RDX: Implement better fetching of image tags

### PR DESCRIPTION
**Pull request by mook-as**
This implements the behaviour specified in #4362 on how to determine which tag to use when installing/uninstalling an extension:
    
- If a tag was supplied, use it.
- Optionally, use the installed version.
- Use the semver-largest tag
- Use "latest"
    
When we attempt to install an extension, we skip looking at the currently installed version.  When we uninstall it, we prefer to select the currently installed version for uninstall.

The commits should be reasonably set up so that reviewing commit-by-commit is doable.

### CODE REVIEW

File: pkg/rancher-desktop/backend/containerClient/mobyClient.ts [hunk 1]
```diff
@@ -177,6 +179,37 @@ export class MobyClient implements ContainerEngineClient {
     }
   }
 
+  async getTags(imageName: string, options?: ContainerBasicOptions) {
```
Comment (ericpromislow) [inline pkg/rancher-desktop/backend/containerClient/mobyClient.ts:182]: Why does the moby client have some log statements in its `getTags` method that aren't in the nerdctl client's `getTags` method?

**Review by ericpromislow — DISMISSED** [review]
I had `aquasec/trivy-docker-extension:0.4.6` installed but specified
`rdctl extension uninstall aquasec/trivy-docker-extension:0.4.9`

and it deleted it. Is this expected?

The issue says:
"If the user specifies a tag, that tag will be used. (If that doesn't exist, it fails.)"

Is this only for installing an extension?

It also deletes if I specify an unrelated tag, like `:0.3`

File: pkg/rancher-desktop/main/extensions/manager.ts [hunk 2]
```diff
@@ -230,36 +232,68 @@ class ExtensionManagerImpl implements ExtensionManager {
...
+    imageName ??= image;
     tag ??= undefined;
 
-    this.extensions[repo] = extGroup;
+    this.extensions[imageName] ??= {};
+    const extGroup = this.extensions[imageName];
+    const preferInstalled = options?.preferInstalled ?? true;
 
     if (tag) {
       // Requested a specific tag; create it if we don't have it.
```
Comment (ericpromislow) [inline pkg/rancher-desktop/main/extensions/manager.ts:247]: Does this comment make sense if we're uninstalling an existing extension? I think I'm not clear on what an `extension` object represents -- more like a wish to install/uninstall a particular extension, than something that is actually installed as an extension?

Comment (mook-as) [reply pkg/rancher-desktop/main/extensions/manager.ts:247]: An `extension` object is, roughly, a wrapper around the metadata (so we can use it to install or uninstall extensions).  For example, there can exist multiple (available) versions of `splatform/epinio-docker-desktop`, so there can be an `extension` object for each; however, only one of them can be installed at a time (that is, we disallow installing multiple versions of an extension simultaneously).

**Review by mook-as — COMMENTED** [review]
Uninstalling the incorrect version is not expected, I'll double check.

**Comment by mook-as** [comment]
Added fix for uninstalling the incorrect version (and included a test for it).

**Comment by mook-as** [comment]
Rebased for merge conflict, still pending review.

**Review by rak-phillip — APPROVED** [review]
I think these changes look good, I'm not running into any issues with what's contained in this PR.

I'm unable to successfully run bats tests, but I don't think this is an issue with any changes represented here.

**Comment by mook-as** [comment]
I'll make the `debugE` changes in a separate PR. Thanks!

**Pull request merged**